src/numeric/decision_rules.jl

```diff
@@ -17,190 +51,169 @@ function filter_mcoeffs(a::Array{Float64,1},b::Array{Float64,1},n::Array{Int64,1
     return coeffs
 end
 
-# Constant Decision Rule
-type ConstantDecisionRule <: AbstractDecisionRule
-    values::Array{Float64,1}
-end
-(dr::ConstantDecisionRule)(x::Array{Float64,1}) = dr.values
-(dr::ConstantDecisionRule)(x::Array{Float64,2}) = repmat(dr.values', size(x, 1), 1)
-(dr::ConstantDecisionRule)(i::Int, x::Union{Vector,Matrix}) =  dr(x)
-(dr::ConstantDecisionRule)(i::Int, j::Int, x::Union{Vector, Matrix}) = dr(x)
 
 
-## Cubic spline decision rules
+#####
+##### 1-argument decision rule
+#####
 
-function set_values!(dr::AbtsractCubicSplineDR, values::Array{Array{Float64,2},1})
-    grid = dr.grid
-    a = grid.min
-    b = grid.max
-    orders = grid.n
-    dr.coefficients = [filter_mcoeffs(a, b, orders, v) for v in values]
-end
+using splines
 
-function set_values!(dr::AbtsractCubicSplineDR, values::Array{Float64,2})
-    set_values!(dr, [values])
+function DecisionRule(grid_exo::EmptyGrid, grid_endo::CartesianGrid, n_x::Int)
+    orders = grid_endo.n
+    coeffs = [zeros(n_x, (orders+2)...)]
+    return DecisionRule{EmptyGrid, CartesianGrid}(grid_exo, grid_endo, n_x, coeffs)
 end
 
-# Decision Rule on Markov Process
-
-type MCDecisionRule <: AbtsractCubicSplineDR
-    process::DiscreteProcess
-    grid::CartesianGrid
-    n_x::Int # number of values
-    coefficients::Array{Array{Float64}}
+function DecisionRule(grid_exo::EmptyGrid, grid_endo::CartesianGrid, values::Array{Array{Float64,2}})
+    n_x = size(values[1], 2)
+    dr = DecisionRule(grid_exo, grid_endo, n_x)
+    set_values!(dr, values)
+    return dr
 end
 
-function DecisionRule(mc::DiscreteMarkovProcess, grid::CartesianGrid, n_x::Int)
-    a = grid.min
-    b = grid.max
-    orders = grid.n
-    n_ms = size(mc.values, 1)
-    coeffs = [zeros(n_x, (orders+2)...) for v in 1:n_ms]
-    return MCDecisionRule(mc, grid, n_x,  coeffs)
+function set_values!(dr::AbstractDecisionRule{EmptyGrid, CartesianGrid}, values::Array{Array{Float64,2},1})
+    a = dr.grid_endo.min
+    b = dr.grid_endo.max
+    orders = dr.grid_endo.n
+    dr.coefficients = [filter_mcoeffs(a, b, orders, v) for v in values]
```

Comment:
Does splines.jl have filtering methods that don't allocate the coefficient vector? If so, given that we allocate memory for the coefficients in the constructor (when we call `zeros`), it would be great to use the memory here.

The same comments holds to other `set_values!` methods